src/azure-cli-core/azure/cli/core/commands/template_create.py

```diff
@@ -40,22 +43,27 @@ def wrapped(namespace):
     else:
         validator = fold_validator
 
-    quotes = '""' if platform.system() == 'Windows' else "''"
-    quote_text = '  Use {} for none.'.format(quotes) if none_flag_value else ''
-    flag_texts = {
-        new_flag_value: '  Creates new by default.{}'.format(quote_text),
-        existing_id_flag_value: '  Uses existing resource.{}'
-                                .format(quote_text),
-        none_flag_value: '  None by default.'
-    }
-    help_text = 'Name or ID of the resource.' + flag_texts[default_value_flag]
+    if not custom_help:
+        quotes = '""' if platform.system() == 'Windows' else "''"
+        quote_text = ' Use {} for none.'.format(quotes) if none_flag_value else ''
+        parent_text = ' If name specified, must also specify {}.'.format(
+            parent_option_flag or parent_name) if parent_name else ''
+        flag_texts = {
+            new_flag_value: '  Creates new by default.{}'.format(quote_text),
+            existing_id_flag_value: '  Uses existing resource.{}{}'
+                                    .format(quote_text, parent_text),
```

Comment:
Not quite familiar with this area. Why we use 2 white-spaces in front of `Create new by default`? Also should the `quote_text` has content like ` or use ... for none`? This way the 2 statements can be connected naturally. 
Also, `Uses existing resource`  should be `The resource must be existing`?